### PR TITLE
new RNC, RNG, XSD

### PR DESCRIPTION
**Pull request by domel**
2007 -> 2005
see
https://www.w3.org/2013/sparql-errata#errata-results-1
https://lists.w3.org/Archives/Public/public-rdf-dawg-comments/2013Aug/0017.html

### CODE REVIEW

File: spec/result.rnc [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+# RELAX NG Schema for SPARQL Query Results XML Format
+#
+# This XML schema is normative.
+# $Id: result.rnc,v 1.1 2007/10/17 14:48:47 eric Exp $
+# 2023-02-16 domel
+#
+#
+
+namespace local = ""
+namespace res = "http://www.w3.org/2005/SPARQL/results#"
```
Comment (afs) [inline spec/result.rnc:11]: Is the right fix to change 2007 to 2005 or vice versa - I'm quite unclear what is the best way to address the errata.

Comment (domel) [reply spec/result.rnc:11]: Yes, it was 2007. I have changed to 2005

**Comment by afs** [comment]
#9 created for changing the deployed schema. This needs to be done (by W3C team).

**Review by afs — APPROVED** [review]
Note an extra step is needed - recorded in #9.

**Comment by ericprud** [comment]
Hi @domel , fabulous contribution but I'm not smart enough to use it. 

I validated the 3rd example from [SPARQL Query Results XML Format§2.3.1](https://www.w3.org/TR/rdf-sparql-XMLres/#vb-results) (eliding the '...'s) and ran into namespace probs:
``` sh
jing ./sparql-results-xml/spec/result.rng ./sparql-results-xml/2.3.1ex3.xml 
/home/eric/checkouts/w3c/./sparql-results-xml/2.3.1ex3.xml:2:56: error: element "sparql" not allowed anywhere; expected element "ns:sparql" (with xmlns:ns="http://www.w3.org/2005/SPARQL/results#")
```
Advice?

**Comment by afs** [comment]
It should be "http://www.w3.org/2005/sparql-results#"

I checked the [rdf-tests](https://github.com/w3c/rdf-tests/) repo and all srx files are "http://www.w3.org/2005/sparql-results#"

Not "http://www.w3.org/2005/SPARQL/results#"

**Review by afs — CHANGES_REQUESTED** [review]
It should be "http://www.w3.org/2005/sparql-results#" not "http://www.w3.org/2005/SPARQL/results#".

**Comment by domel** [comment]
Now should be OK. I was check by `xmllint`

**Review by afs — APPROVED** [review]

**Review by rubensworks — APPROVED** [review]

File: spec/result.rnc [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+start = sparql
+
+xmlLang = attribute xml:lang { text }
+
+# Using xsi namespaced terms such as xsi:schemaLocation
+# is tricky. as Relax requires them to be given when used.  However,
+# WXS has this knowledge built in and validators complain when they
+# are explicitly listed without a schemaLocation (which is forbidden by
+# the WXS spec)..
```
Comment (TallTed) [inline spec/result.rnc:24]:
```suggestion
# is tricky, as Relax requires them to be given when used.  However,
# WXS has this knowledge built in and validators complain when they
# are explicitly listed without a schemaLocation (which is forbidden by
# the WXS spec).
```

File: spec/result.rng [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+  </start>
+  <define name="xmlLang">
+    <attribute name="xml:lang"/>
+  </define>
+  <!--
+    Using xsi namespaced terms such as xsi:schemaLocation
+    is tricky. as Relax requires them to be given when used.  However,
+    WXS has this knowledge built in and validators complain when they
+    are explicitly listed without a schemaLocation (which is forbidden by
+    the WXS spec)..
```
Comment (TallTed) [inline spec/result.rng:24]:
```suggestion
    is tricky, as Relax requires them to be given when used.  However,
    WXS has this knowledge built in and validators complain when they
    are explicitly listed without a schemaLocation (which is forbidden by
    the WXS spec).
```

File: spec/result.xsd [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+  <xs:import namespace="http://www.w3.org/XML/1998/namespace" schemaLocation="http://www.w3.org/2001/xml.xsd"/>
+  <xs:attributeGroup name="xmlLang">
+    <xs:attribute ref="xml:lang" use="required"/>
+  </xs:attributeGroup>
+  <!--
+    Using xsi namespaced terms such as xsi:schemaLocation
+    is tricky. as Relax requires them to be given when used.  However,
+    WXS has this knowledge built in and validators complain when they
+    are explicitly listed without a schemaLocation (which is forbidden by
+    the WXS spec)..
```
Comment (TallTed) [inline spec/result.xsd:22]:
```suggestion
    is tricky, as Relax requires them to be given when used.  However,
    WXS has this knowledge built in and validators complain when they
    are explicitly listed without a schemaLocation (which is forbidden by
    the WXS spec).
```

**Review by TallTed — CHANGES_REQUESTED** [review]

**Comment by rubensworks** [comment]
@domel Do you agree with the suggested changes from @TallTed?

**Comment by domel** [comment]
Done

**Pull request merged**